c_api/include/taichi/cpp/taichi.hpp

```diff
@@ -102,10 +107,36 @@ class Memory {
     destroy();
     runtime_ = detail::move_handle(b.runtime_);
     memory_ = detail::move_handle(b.memory_);
+    size_ = std::exchange(b.size_, 0);
     should_destroy_ = std::exchange(b.should_destroy_, false);
     return *this;
   }
 
+  void *map() const {
+    return ti_map_memory(runtime_, memory_);
+  }
+  void unmap() const {
+    ti_unmap_memory(runtime_, memory_);
+  }
+
+  inline void read(void *dst, size_t size) const {
```

Comment:
I wonder if read/write should go through staging buffer instead of depending on map/unmap? As some devallocs aren't allocated with host_read/write and map/unmap returns nullptr?

Reply:
Will ensure every map access is valid by staging buffer